ci-operator/step-registry/rosa/cluster/provision/rosa-cluster-provision-commands.sh

```diff
@@ -78,8 +78,21 @@ else
   fi
   subfix=$(openssl rand -hex 2)
   CLUSTER_NAME=${CLUSTER_NAME:-"$prefix-$subfix"}
+  # For long cluster name enabled, append a 39 chars long random string to cluster name
+  # Max possible prefix length is 9, max possible subfix length is 4, hyppen is used 2 times, random string of length 39
+  # (9 + 4 + 2 + 39 = 54)
+  if [[ "$LONG_CLUSTER_NAME_ENABLED" == "true" ]]; then
+    long_name_prefix=$(openssl rand -hex 20 | tr -dc 'a-f' | cut -c -39)
+    CLUSTER_NAME=${CLUSTER_NAME:-"$prefix-$subfix-$long_name_prefix"}
+  fi
+  echo "${CLUSTER_NAME}" > "${SHARED_DIR}/cluster-name"
+fi
+
+DOMAIN_PRIFIX_SWITCH=""
+if [[ "$DOMAIN_PRIFIX_ENABLED" == "true" ]]; then
+  DOMAIN_PRIFIX=${CLUSTER_NAME:-"$prefix-$subfix"}
+  DOMAIN_PRIFIX_SWITCH="--domain-prefix $DOMAIN_PREFIX"
 fi
-echo "${CLUSTER_NAME}" > "${SHARED_DIR}/cluster-name"
```

Comment:
can you please provide some more information on why should calculate the `cluster_name_len` and afer the calculation what will we do with that value?

Reply:
The cluster_name_len could be more flexible than hardcode to 39.

Reply:
using a flexible cluster length of cluster name rather than 39, the length of this will dynamically vary between the range (0 to 39)  
`head -c $(( RANDOM % 40 )) /dev/urandom | tr -dc 'a-z0-9' | head -c 39`